src/app.rs

```diff
@@ -333,15 +273,160 @@ impl<'a, 'b, T> ApplicationBuilder<'a, 'b, T> {
+    ///
+    /// # Returns
+    ///
+    /// This function returns the ApplicationBuilder after modifying it.
+    pub fn with_frame_limit(mut self, strategy: FrameRateLimitStrategy, max_fps: u32) -> Self {
+        self.frame_limiter = FrameLimiter::new(strategy, max_fps);
+        self
+    }
+
+    /// Sets the maximum frames per second of this game, based on the given config.
+    ///
+    /// # Parameters
+    ///
+    /// `config`: the frame limiter config
+    ///
+    /// # Returns
+    ///
+    /// This function returns the ApplicationBuilder after modifying it.
+    pub fn with_frame_limit_config(mut self, config: FrameRateLimitConfig) -> Self {
+        self.frame_limiter = FrameLimiter::from_config(config);
+        self
+    }
+
+    /// Sets the duration between fixed updates, defaults to one sixtieth of a second.
+    ///
+    /// # Parameters
+    ///
+    /// `duration`: The duration between fixed updates.
+    ///
+    /// # Returns
+    ///
+    /// This function returns the ApplicationBuilder after modifying it.
+    pub fn with_fixed_step_length(mut self, duration: Duration) -> Self {
+        self.time.set_fixed_time(duration);
+        self
+    }
+
+    /// Build an `Application` object using the `ApplicationBuilder` as configured.
+    ///
+    /// # Returns
+    ///
+    /// This function returns an Application object wrapped in the Result type.
+    ///
+    /// # Errors
+    ///
+    /// This function currently will not produce an error, returning a result
+    /// type was strictly for future possibilities.
+    ///
+    /// # Notes
+    ///
+    /// If the "profiler" feature is used, this function will register the thread
+    /// that executed this function as the "Main" thread.
+    ///
+    /// # Examples
+    ///
+    /// See the [example show for `ApplicationBuilder::new()`](struct.ApplicationBuilder.html#examples)
+    /// for an example on how this method is used.
+    pub fn build<'a>(self) -> Result<Application<'a, T>> where S: State<T> + 'a {
+        trace!("Entering `ApplicationBuilder::build`");
+
+        #[cfg(feature = "profiler")]
+        register_thread_with_profiler("Main".into());
+        #[cfg(feature = "profiler")]
+        profile_scope!("new");
+
+        Ok(Application {
+            data: self.data.unwrap(),
+            states: StateMachine::new(self.initial_state),
+            time: self.time,
+            frame_limiter: self.frame_limiter,
+            stopwatch: Stopwatch::new(),
+        })
+    }
+}
+
+/// Default game data
+pub struct DefaultGameData<'a, 'b> {
+    world: World,
+    dispatcher: Dispatcher<'a, 'b>,
+    events_reader_id: ReaderId<Event>,
+}
+
+impl<'a, 'b> DefaultGameData<'a, 'b> {
+    /// Build
+    pub fn build<P: AsRef<Path>>(path: P) -> Result<DefaultGameDataBuilder<'a, 'b>> {
+        DefaultGameDataBuilder::new(path)
+    }
+
+    /// Get events
+    pub fn get_window_events<I>(&mut self) -> Vec<Event> {
```

Comment:
I'd be very much in favor of calling this `poll_window_events` since `get_` makes the impression of a simple getter.

Reply:
That's a better name yeah